source/tests/system/nixnet_lin_driver_api_tests.cpp

```diff
@@ -0,0 +1,167 @@
+
+using namespace nixnet_grpc;
+namespace client = nixnet_grpc::experimental::client;
+namespace pb = google::protobuf;
+using namespace ::testing;
+using namespace ::nixnet_utilities;
+using nlohmann::json;
+
+namespace ni {
+namespace tests {
+namespace system {
+namespace {
+
+typedef pb::uint8 u8;
+typedef pb::uint32 u32;
+typedef pb::uint64 u64;
+typedef double f64;
+
+class NiXnetLINDriverApiTests : public ::testing::Test {
+ protected:
+  NiXnetLINDriverApiTests()
+      : device_server_(DeviceServerInterface::Singleton()),
+        stub_(NiXnet::NewStub(device_server_->InProcessChannel()))
+  {
+    device_server_->ResetServer();
+  }
+  virtual ~NiXnetLINDriverApiTests() {}
+
+  void TearDown() override
+  {
+    device_server_->ResetServer();
+  }
+
+  template <typename TService>
+  std::unique_ptr<typename TService::Stub> create_stub()
+  {
+    return TService::NewStub(device_server_->InProcessChannel());
+  }
+
+  std::unique_ptr<NiXnet::Stub>& stub()
+  {
+    return stub_;
+  }
+
+ private:
+  DeviceServerInterface* device_server_;
+  std::unique_ptr<NiXnet::Stub> stub_;
+};
+
+TEST_F(NiXnetLINDriverApiTests, ConvertFramesToFromSignalsFromExample_FetchData_DataLooksReasonable)
+{
+  constexpr auto NUM_FRAMES = 2;
+  constexpr auto NUM_SIGNALS = 2;
+  std::vector<nixnet_grpc::FrameBufferRequest> frames;
+  auto session = EXPECT_SUCCESS(client::create_session(stub(), "NIXNET_exampleLDF", "Cluster", "SlaveSignal1_U16,MasterSignal1_U16", "", CREATE_SESSION_MODE_SIGNAL_CONVERSION_SINGLE_POINT)).session();
+  nixnet_grpc::FrameRequest* frame = new nixnet_grpc::FrameRequest();
+  frame->set_timestamp(0);
+  frame->add_flags(FrameFlags::FRAME_FLAGS_UNSPECIFIED);
+  frame->set_identifier(4);
+  frame->set_type(FrameType::FRAME_TYPE_LIN_DATA);
+  frame->set_payload("\0\1\2\3\4\5\6\7");
+  frames.push_back(nixnet_grpc::FrameBufferRequest());
+  frames.back().set_allocated_lin(frame);
+  nixnet_grpc::FrameRequest* frame2 = new nixnet_grpc::FrameRequest();
+  frame2->set_timestamp(0);
+  frame2->add_flags(FrameFlags::FRAME_FLAGS_UNSPECIFIED);
+  frame2->set_identifier(2);
+  frame2->set_type(FrameType::FRAME_TYPE_LIN_DATA);
+  frame2->set_payload("\0\1\2\3\4\5\6\7");
+  frames.push_back(nixnet_grpc::FrameBufferRequest());
+  frames.back().set_allocated_lin(frame2);
+
+  auto convert_frames_to_signals_single_point_response = EXPECT_SUCCESS(client::convert_frames_to_signals_single_point(stub(), session, NUM_SIGNALS, frames));
+  std::vector<f64> value_buffer_copy(convert_frames_to_signals_single_point_response.value_buffer().begin(), convert_frames_to_signals_single_point_response.value_buffer().end());
+  auto convert_signals_to_frames_single_point_response = EXPECT_SUCCESS(client::convert_signals_to_frames_single_point(stub(), session, value_buffer_copy, NUM_FRAMES, 8, Protocol::PROTOCOL_LIN));
+  EXPECT_SUCCESS(client::clear(stub(), session));
+
+  EXPECT_EQ(2, convert_frames_to_signals_single_point_response.timestamp_buffer().size());
```

Comment:
Have you been able to look at the frames in the debugger at least? This should be the same data that is populated in the frames above, right?

Reply:
Yes, they were same.